modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java

```diff
@@ -196,14 +196,12 @@ else if (Objects.equals(type, "list")) {
 				List<KeyValuePair> keyValuePairs = new ArrayList<>();
 
 				for (int i = 0; i < values.size(); i++) {
+					Object key = keys.get(i);
+
+					Object value = values.get(i);
+
 					keyValuePairs.add(
-						new KeyValuePair(
-							keys.get(
-								i
-							).toString(),
-							values.get(
-								i
-							).toString()));
+						new KeyValuePair(key.toString(), value.toString()));
```

Comment:
@cleydyr would it be possible to use

```
keyValuePairs.add(
	new KeyValuePair(
		String.valueOf(keys.get(i)),
		String.valueOf(values.get(i))));
```

instead?

Reply:
Sure. It would be pretty much the same thing, only with a null-check added. I'll change it.